sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallScheduler.java

```diff
@@ -87,14 +108,26 @@ private Plan getPlan() {
 
         Phase resourcePhase = new DefaultPhase(RESOURCE_PHASE, taskSteps, new ParallelStrategy<>(),
                 Collections.emptyList());
+        phases.add(resourcePhase);
+
+        if (secretsClient.isPresent()) {
+            Step tlsCleanupStep = new TLSCleanupStep(
```

Comment:
1. It may be a valid request to keep the TLS artifacts. Would you like to implement another scheduler flag which would control such a behavior?

2. In case of any failure the TLS certificates would be reused (if artifacts are complete in secret store) or overwritten when existing values couldn't be used. In other words, failed uninstall wouldn't prevent reinstalling the service.